protostar/cheatable_starknet/controllers/expect_events_controller.py

```diff
@@ -0,0 +1,166 @@
+        if not expected_events or not should_accept_event_matching(
+            expected_event=expected_events[0], emitted_event=emitted_event
+        ):
+            event_matchings.append(SkippedEventMatching(emitted_event=emitted_event))
+        else:
+            event_matchings.append(
+                AcceptedEventMatching(
+                    emitted_event=emitted_event,
+                    expected_event=expected_events[0],
+                )
+            )
+            expected_events.pop(0)
+    failed_event_matchings = [
+        FailedEventMatching(expected_event) for expected_event in expected_events
+    ]
+    return EventMatchingResult(
+        event_matchings=[*event_matchings, *failed_event_matchings],
+    )
+
+
+def should_accept_event_matching(expected_event: Event, emitted_event: Event) -> bool:
+    return (
+        expected_event.key == emitted_event.key
+        and (expected_event.data is None or (expected_event.data == emitted_event.data))
+        and expected_event.from_address == emitted_event.from_address
+    )
+
+
+class ExpectEventsMismatchReportedException(ReportedException):
+    def __init__(
+        self, event_matching_result: EventMatchingResult, *args: object
+    ) -> None:
+        super().__init__(*args)
+        self.event_matching_result = event_matching_result
+
+    def __reduce__(self):
+        return (
+            type(self),
+            (self.event_matching_result,),
+            self.__getstate__(),
+        )
+
+    def __str__(self) -> str:
+        formatted_event_matches_lines = self._get_event_matches_formatted_lines()
+        return "Expect Events mismatch\n  " + (
+            "\n  ".join(formatted_event_matches_lines)
+        )
+
+    def _get_event_matches_formatted_lines(self):
+        lines: list[str] = []
+        for event_matching in self.event_matching_result.event_matchings:
+            if isinstance(event_matching, AcceptedEventMatching):
+                lines.append(self._format_accepted_event_matching(event_matching))
+            elif isinstance(event_matching, SkippedEventMatching):
+                lines.append(self._format_skipped_event_matching(event_matching))
+            elif isinstance(event_matching, FailedEventMatching):
+                lines.append(self._format_failed_event_matching(event_matching))
+            else:
+                assert False, f"Unexpected event_matching: {event_matching}"
+        return lines
+
+    def _format_accepted_event_matching(self, event_matching: AcceptedEventMatching):
+        return self._format_event(event=event_matching.emitted_event, prefix="[pass]")
+
+    def _format_failed_event_matching(self, event_matching: FailedEventMatching):
+        return self._format_event(event=event_matching.expected_event, prefix="[fail]")
+
+    def _format_skipped_event_matching(self, event_matching: SkippedEventMatching):
+        return self._format_event(event=event_matching.emitted_event, prefix="[skip]")
+
```

Comment:
```suggestion
    def _get_event_matches_formatted_lines(self):
        lines: list[str] = []
        for event_matching in self.event_matching_result.event_matchings:
            if isinstance(event_matching, AcceptedEventMatching):
                lines.append(self._format_event(event=event_matching.emitted_event, prefix="[pass]"))
            elif isinstance(event_matching, SkippedEventMatching):
                lines.append(self._format_event(event=event_matching.emitted_event, prefix="[skip]"))
            elif isinstance(event_matching, FailedEventMatching):
                lines.append(self._format_event(event=event_matching.expected_event, prefix="[fail]"))
            else:
                assert False, f"Unexpected event_matching: {event_matching}"
        return lines
```
redundant methods:
- `_format_accepted_event_matching`
- `_format_failed_event_matching`
- `_format_skipped_event_matching`

Reply:
[remove functions](https://github.com/software-mansion/protostar/pull/1412/commits/4b29d7257b3235d3c6dc1ef9943d35230208f526)